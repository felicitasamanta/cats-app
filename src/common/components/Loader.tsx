import loader from "@common/assets/loading.gif";
import classes from "@common/styles/Loader.module.css";
const Loader = () => {
  return (
    <div className={classes.loader}>
      <div>
        <img src={loader} className={classes.img} alt="Loader" />
      </div>
    </div>
  );
};
export { Loader };
