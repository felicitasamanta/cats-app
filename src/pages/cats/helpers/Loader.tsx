import loader from "./loading.gif";
import classes from "./Loader.module.css";
const Loader = () => {
  return (
    <div className={classes.loader}>
      <div>
        <h1>Loading...</h1>
        <img src={loader} alt="Loader" />
      </div>
    </div>
  );
};
export { Loader };
