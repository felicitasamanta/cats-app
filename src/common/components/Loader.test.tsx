import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { Loader } from "./Loader";

describe("Renders loader", () => {
  it("should render correctly", () => {
    expect(render(<Loader />)).toMatchSnapshot();
  });

  it("should render loader image", () => {
    render(<Loader />);
    const displayedImage = document.querySelector("img") as HTMLImageElement;
    expect(displayedImage.getAttribute("src")).toContain("loading.gif");
  });
});
