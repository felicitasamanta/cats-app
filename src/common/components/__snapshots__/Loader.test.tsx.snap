// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders loader > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_loader_0fe09f"
      >
        <div>
          <img
            alt="Loader"
            class="_img_0fe09f"
            src="/src/common/assets/loading.gif"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_loader_0fe09f"
    >
      <div>
        <img
          alt="Loader"
          class="_img_0fe09f"
          src="/src/common/assets/loading.gif"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
